translation/message_format.rst

```diff
@@ -0,0 +1,436 @@
+        ? 0
+        : ((($number % 10 >= 2)
+            && ($number % 10 <= 4)
+            && (($number % 100 < 10)
+            || ($number % 100 >= 20)))
+                ? 1
+                : 2
+    );
+
+In order to facilitate this, the possible cases in the ``plural`` function are
+also different for each language. For instance, Russian has ``one``, ``few``,
+``many`` and ``other``, while English has only ``one`` and ``other``. The full
+list of possible cases can be found in Unicode's `Language Plural Rules`_
+document. By prefixing with ``=``, you can match exact values (like ``0`` in
+the above example).
+
+Usage of this string is the same as with variables and select::
+
+    // ...
+
+    // prints "There is one apple..."
+    echo $translator->trans('nr_of_apples', ['apples' => 1]);
+
+    // prints "There are 23 apples!"
+    echo $translator->trans('nr_of_apples', ['apples' => 23]);
+
+.. note::
+
+    You can also set an ``offset`` variable to determine whether the
+    pluralization should be offset (e.g. in sentences like ``You and # other people``
+    / ``You and # other person``). 
+
+.. tip::
+
+    When combining the ``select`` and ``plural`` functions, try to still have
+    ``select`` as outermost function:
+
+    .. code-block:: text
+
+		{gender_of_host, select, 
+            female {
+                {num_guests, plural, offset:1 
+                =0    {{host} does not give a party.}
+                =1    {{host} invites {guest} to her party.}
+                =2    {{host} invites {guest} and one other person to her party.}
+                other {{host} invites {guest} and # other people to her party.}}
+            }
+            male {
+                {num_guests, plural, offset:1 
+                =0    {{host} does not give a party.}
+                =1    {{host} invites {guest} to his party.}
+                =2    {{host} invites {guest} and one other person to his party.}
+                other {{host} invites {guest} and # other people to his party.}}
+            }
+            other {
+                {num_guests, plural, offset:1 
+                =0    {{host} does not give a party.}
+                =1    {{host} invites {guest} to their party.}
+                =2    {{host} invites {guest} and one other person to their party.}
+                other {{host} invites {guest} and # other people to their party.}}
+            }
+        }
+
+Other Placeholder Functions
+---------------------------
+
+Besides these, the MessageFormat comes with a couple other interesting functions.
+
+Ordinal
+~~~~~~~
+
+Similair to ``plural``, ``selectordinal`` allows you to use numbers as ordinal scale:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # translations/messages.en.yaml
+        finish_place: >
+            You finished {place, selectordinal,
+                one {#st}
+                two {#nd}
+                few {#rd}
+                other {#th}
+            }!
+
```

Comment:
I agree. Using "1, 2, 3" makes it easier to understand than using "one, two, few"  ("few" specially is so vague that it's confusing).

Reply:
Well, actually they're not the same and wasn't my point :P

My suggestion is to split the example; the first one can be simplified to just using 'ordinal' and let ICU handle all that formatting. A better example for selectordinal is to use something like my example. Although the other-case should perhaps be adjusted to something like this:
`other {didn't finish in the top three, but on # place}` (haven't tested that this works though). It could otherwise also be implemented using 'count'.

Btw not just 'few' can be confusing, the one and two may also be confusing. Some languages apply the one and two rule to any number ending in 1 and 2 (i.e. 22 => twenty-second), for either or both of 'count' and 'ordinal'.

Reply:
Yeah, keywords might be confusing (but also not that much imo). But that's how the standard works, advising a bad practice to use `=...` is not a good idea imo.

I wasn't aware of the `ordinal` function. As I want to show how this `selectordinal` works, I've now extended this example by also adding a line showing the "simpler" `ordinal` case used for the same message. I can't think of a good example showing "one", "two", "few" in `selectordinal` without it being simplified to `ordinal`.